packages/cozy-client/src/hooks/useClientErrors.spec.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import useClientErrors from './useClientErrors'
+import { renderHook, act } from '@testing-library/react-hooks'
+import CozyProvider from '../Provider'
+import { shallow } from 'enzyme'
+import { FetchError } from 'cozy-stack-client'
+
+function createCozyClient() {
+  return {
+    on: jest.fn(),
+    removeListener: jest.fn()
+  }
+}
+
+function createWrapper(client = createCozyClient()) {
+  function Wrapper({ children }) {
+    return <CozyProvider client={client}>{children}</CozyProvider>
+  }
+  return Wrapper
+}
+
+function renderWrappedHook(params, client) {
+  const wrapper = createWrapper(client)
+  return renderHook(() => useClientErrors(), { wrapper })
+}
+
+function wrappedShallow(tree, client) {
+  return shallow(tree, { wrappingComponent: createWrapper(client) })
+}
+
+describe('useClientErrors', () => {
+  it('registers an `error` handler in client', done => {
+    const client = createCozyClient()
+    client.on.mockImplementation(name => name === 'error' && done())
+    renderWrappedHook(undefined, client)
+  })
```

Comment:
This seems like testing for an implementation detail.

What is important here is that ClientErrors contains errors if cozy-client has an error, the mechanism through which this happens should not be tested IMHO.

Reply:
Hum… may be related to https://github.com/cozy/cozy-client/pull/676#discussion_r400305385

I see the cozyClient as an injectable dependency, with an external public interface, not some internal behaviour of the hook.